### DISCUSSION TITLE
Alerting: When adding an alarm to a group the status of the whole group is reset

### DISCUSSION THREAD

**Discussion by Jessimon** [grafana/grafana#52314] [Alerting]
Hello Grafana,

I am using Grafana 9.0.2 with unified alerting.

Problem:
When i add an alarm to a group that has firing alarm, the firing alarms are reset to resolve.

Expectation:
Adding an alarm to a group should not affect the status of other alarms in that group.

Reproduce:
Create an alarm that always fires A=1 and $A < 1 in a new group.
Then add a new alarm to the same group.
The first alarm will go from firing to resolved.

**Environment**:
- Grafana version: 9.0.2
- Data source type & version: Prometheus 2.35.0
- OS Grafana is installed on: Debian 11
- User OS & Browser: Windows 10 with Firefox

Thank you.

**Comment by JohnnyQQQQ** [comment]
Thanks for opening this Issue. This is expected behavior for alert rule groups. Rule groups are an atomic unit of groups that will (in the future) be executed in a certain order. Rules within a rule group can use values from previous executed rules in this group. This is partly why we need to reset the state on any change to the rule group.

This is pretty new and we are still looking at how we can improve it.

**Comment by yuri-tceretian** [comment]
PR https://github.com/grafana/grafana/pull/66531 will fix this

**Comment by yuri-tceretian** [comment]
Fixed in https://github.com/grafana/grafana/pull/66531 backported to 9.5.x, and will be released in 9.5.2
